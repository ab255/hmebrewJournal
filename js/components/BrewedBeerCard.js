import React, { Component } from 'react';
import {
  Image,
  Text,
  TouchableHighlight,
  StyleSheet,
  View,
} from 'react-native';
import moment from 'moment';

const BrewedBeerCard = ({ beerName, beerType, brewDate, onPress }) => {
  return (
    <TouchableHighlight onPress={onPress}>
      <View style={styles.brewedBeerCard}>
        <View>
          <Text style={styles.beerCardTitle}>{beerName}</Text>
          <Text style={styles.beerCardType}>{beerType}</Text>
          <Text style={styles.beerCardDate}>{moment(brewDate).format('MMMM Do YYYY')}</Text>
        </View>
        <View>
          <Image source={require('../icons/arrow-right-card.png')} />
        </View>
      </View>
    </TouchableHighlight>
  )
}

const styles = StyleSheet.create({
  brewedBeerCard: {
    flexDirection: 'row',
    backgroundColor: '#FCF0C8',
    borderBottomWidth: 1,
    borderColor: '#630A10',
    marginBottom: 2,
    padding: 10,
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  beerCardTitle: {
    color: '#630A10',
    fontSize: 20,
    fontWeight: 'bold',
  },
  beerCardType: {
    color: '#630A10',
  },
  beerCardDate: {
    color: '#630A10',
  },
})

module.exports = BrewedBeerCard;
