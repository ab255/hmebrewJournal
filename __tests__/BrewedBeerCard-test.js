import 'react-native';
import React from 'react';
import BrewedBeerCard from '../js/components/BrewedBeerCard.js';

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <BrewedBeerCard />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
