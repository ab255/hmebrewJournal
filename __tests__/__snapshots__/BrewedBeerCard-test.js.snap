exports[`test renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FCF0C8",
        "borderBottomWidth": 1,
        "borderColor": "#630A10",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 2,
        "padding": 10,
      }
    }>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#630A10",
            "fontSize": 20,
            "fontWeight": "bold",
          }
        } />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#630A10",
          }
        } />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#630A10",
          }
        }>
        November 10th 2016
      </Text>
    </View>
    <View>
      <Image
        source={1} />
    </View>
  </View>
</View>
`;
